pkg/services/datasources/service/store.go

```diff
@@ -102,22 +104,35 @@ func (ss *SqlStore) GetAllDataSources(ctx context.Context, query *datasources.Ge
 }
 
 // GetDataSourcesByType returns all datasources for a given type or an error if the specified type is an empty string
-func (ss *SqlStore) GetDataSourcesByType(ctx context.Context, query *datasources.GetDataSourcesByTypeQuery) ([]*datasources.DataSource, error) {
+func (ss *sqlStore) GetDataSourcesByType(ctx context.Context, query *datasources.GetDataSourcesByTypeQuery) ([]*datasources.DataSource, error) {
```

Comment:
Did you consider changing this so that the query argument can take a list of types instead and not having to include plugin management in the store?

Reply:
I mean so that datasource service still takes a single type as argument, but can send multiple arguments to store given aliases exists. Maybe that doesn't change a lot? I was under the impression that datasource service already had plugin dependency, but apparently not

Reply:
The pros is that it's easier to implement. The cons is that caller needs to worry to resolve the plugin instance before calling the function. See at the alternative: https://github.com/grafana/grafana/pull/76361. Both options sounds good to me. WDYT?

Reply:
ah, you mean that wrapper. Okay, that can be a third option. I can investigate that.

Reply:
Too many options :smile:

Reply:
see the alternative: https://github.com/grafana/grafana/pull/76637 It's definetly less changes but I am not sure if doing the alias resolution in that intermediate step is the best solution. Let me know what you think!